base/terminfo.jl

```diff
@@ -337,6 +337,10 @@ function ttyhastruecolor()
         get(current_terminfo, :RGB, false) || get(current_terminfo, :Tc, false) ||
         (haskey(current_terminfo, :setrgbf) && haskey(current_terminfo, :setrgbb)) ||
         @static if Sys.isunix() get(current_terminfo, :colors, 0) > 256 else false end ||
+        @static if Sys.iswindows()
+            is_windows_console = ccall((:GetConsoleWindow, "kernel32"), stdcall, UInt32, () ) != UInt32(0) # See <https://learn.microsoft.com/en-us/windows/console/getconsolewindow>
+            Sys.windows_version() ≥ v"10.0.14931" && is_windows_console # See <https://devblogs.microsoft.com/commandline/24-bit-color-in-the-windows-console/>
+        else false end ||
```

Comment:
```suggestion
        (Sys.isunix() && get(current_terminfo, :colors, 0) > 256) ||
        (Sys.iswindows() &&
            let is_windows_console = ccall((:GetConsoleWindow, "kernel32"), stdcall, Ptr{Cvoid}, ()) != C_NULL
            Sys.windows_version() ≥ v"10.0.14931" && is_windows_console # See <https://devblogs.microsoft.com/commandline/24-bit-color-in-the-windows-console/>
        ) ||
```

HWND is a HANDLE which is a PVOID which is a void*

Also `@static` really is only for syntactic issues (which used to include `ccall`, but does not anymore), so a regular conditional is fine here. I think `||` and `&&` is a bit more common than `if foo bar else false end` style also

Reply:
I don't see from that documentation how `GetConsoleWindow` returns a useful result here

Reply:
In particular, terminal emulators will still typically have a `GetConsoleWindow`, even if they are screen-scraping it to re-render it to something else

Reply:
Ok, I left only the check against Windows version.